middleware/redisClientAsync.js

```diff
@@ -13,23 +15,19 @@ redisClient.on("err", err => {
 const redisGetAsync = promisify(redisClient.get).bind(redisClient);
 const redisSetAsync = promisify(redisClient.setex).bind(redisClient);
 
-const asyncSetValue = (id, value) => {
-  const key = `${redisKeyPrefix}-${id}`; 
-  const doc = {
-      value: value,
-      status: 'Current'
-  }
-  redisSetAsync(key, config.cacheExpiration, doc);
+const asyncSetValue = (value, expiration) => {
+  const id = randomUUID;
+  const key = `${redisKeyPrefix}-${id}`;
+
+  redisSetAsync(key, expiration, value);
+  return key;
 };
 
 const asyncGetValue = id => {
     const key = `${redisKeyPrefix}-${id}`;
     const val = await redisGetAsync(key);
-    const invalidated = {
-        value: '',
-        status: 'Consumed'
-    }
-    redisSetAsync(key, config.cacheExpiration, invalidated);
+
+    redisSetAsync(key, config.cacheExpiration, '');
```

Comment:
Looks like it should be `redisGetAsync`.

Reply:
No, that's the call that's overwriting the message with a blank after retrieving it